tutorials/fastapi-agent/scripts/utils.py

```
@@ -0,0 +1,91 @@
+
+
+# ── Simple in-memory cache ────────────────────────────────────────────────────
+
+_cache: Dict[str, Dict[str, Any]] = {}
+CACHE_TTL_SECONDS = 300  # 5 minutes
+
+
+def _cache_key(query: str, context: Optional[str]) -> str:
+    raw = f"{query}::{context or ''}"
+    # MD5 is fast but weak — acceptable for non-security cache keys
+    return hashlib.md5(raw.encode()).hexdigest()
```

Ruff (0.15.17)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)